### PR TITLE
#2589 #2618 

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#2589 Port Dev15 references and setup changes from 0.5 to master 
#2618 Clean up settings storage/persistence code 

...and fix Help tests

### CODE REVIEW

File: src/Debugger/Impl/project.14.0.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 ﻿{
   "dependencies": {
     "MicroBuild.Core": "0.2.0",
-    "Microsoft.Tpl.Dataflow": "4.5.24",
     "Microsoft.VisualStudio.Debugger.Interop.10.0": "10.0.30319",
     "Microsoft.VisualStudio.Debugger.Interop.14.0": "14.3.25407",
-    "Microsoft.VisualStudio.SDK.EmbedInteropTypes": "14.1.4",
+    "Microsoft.VisualStudio.SDK.EmbedInteropTypes": "14.1.2",
```
Comment (MikhailArkhipov) [inline src/Debugger/Impl/project.14.0.json]: Oops, need to put back

File: src/Package/TestApp/Help/HelpWindowTest.cs [hunk 2]
```diff
@@ -68,36 +62,41 @@ public class HelpWindowTest : HostBasedInteractiveTest {
...
+            await UIThreadHelper.Instance.InvokeAsync(() => {
                 IHTMLElement2 body = component.Browser.Document.Body.DomElement as IHTMLElement2;
                 color = body.currentStyle.backgroundColor as string;
             });
+
             return color;
         }
+
+        private async Task WaitForReadyAndRenderedAsync(RHostClientHelpTestApp clientApp) {
+            await UIThreadHelper.Instance.InvokeAsync(() => DoIdle(500));
```
Comment (AlexanderSher) [inline src/Package/TestApp/Help/HelpWindowTest.cs]: In concurrent environment, such delays work not as during single run (UI thread will be immediately taken by another test).
What do you want to achieve?

Comment (MikhailArkhipov) [inline src/Package/TestApp/Help/HelpWindowTest.cs]: Browser needs UI thread time to complete rendering. It is old IE it does things via PostMessage

File: src/Package/TestApp/Help/RHostClientHelpTestApp.cs [hunk 3]
```diff
@@ -5,36 +5,57 @@
...
+using Microsoft.Common.Core;
 using Microsoft.R.Components.Help;
 using Microsoft.R.Host.Client.Test.Script;
 using Microsoft.UnitTests.Core.Threading;
 
 namespace Microsoft.VisualStudio.R.Interactive.Test.Help {
     class RHostClientHelpTestApp : RHostClientTestApp {
-        IHelpVisualComponent _component;
+        private IHelpVisualComponent _component;
+        private TaskCompletionSource<bool> _tcs = new TaskCompletionSource<bool>();
```
Comment (AlexanderSher) [inline src/Package/TestApp/Help/RHostClientHelpTestApp.cs]: If you want to asynchronously wait for event to happen, you can define `EventTaskSource` for it (for example, `EventTaskSources.IRSession.Mutated`)

Comment (MikhailArkhipov) [inline src/Package/TestApp/Help/RHostClientHelpTestApp.cs]: Actually I don't use that field anymore, went wit regular event. But I'll look into it

File: src/Package/TestApp/Help/HelpWindowTest.cs (hunk 2 above)
Comment (MikhailArkhipov) [inline src/Package/TestApp/Help/HelpWindowTest.cs]: I can try. Idea was to wait outside of UI thread to give browser time and have UI thread unblocked for rendering. Would sitting inside InvokeAsync block UI thread?

Comment (AlexanderSher) [inline src/Package/TestApp/Help/HelpWindowTest.cs]: There is a `UIThreadTools` helper class that you can use for it, but I don't see big value in changing it.
The only thing I'd ask you to change is to use `BeCompletedAsync`:
```{csharp}
await UIThreadHelper.Instance.InvokeAsync(() => DoIdle(500));
await clientApp.Ready.Should().BeCompletedAsync(5000);
await UIThreadHelper.Instance.InvokeAsync(() => DoIdle(500));
```


---
In reply to: [86628786](https://github.com/Microsoft/RTVS/pull/2642#discussion_r86628786) [](ancestors = 86628786)

**Comment by MikhailArkhipov** [comment]
Anything else to fix?

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**